Source/SpectrumAnalysis/KTSequentialTrackFinder.cc

```diff
@@ -329,225 +341,312 @@ namespace Katydid
-                         bool anyPointCondition = std::abs(pointIt->fFrequency - (lineIt->GetEndFrequency() + lineIt->GetSlope()*(pointIt->fTimeInAcq - lineIt->GetEndTimeInAcq()))) < fFrequencyAcceptance;
-                         bool secondPointCondition = std::abs(pointIt->fFrequency - (lineIt->GetEndFrequency() + lineIt->GetSlope()*(pointIt->fTimeInAcq - lineIt->GetEndTimeInAcq()))) < fInitialFrequencyAcceptance;
-
-                         // if point matches this line: insert
-                         if (timeCondition and anyPointCondition)
-                         {
-                             KTDEBUG(stflog, "Matching conditions fullfilled");
-                             lineIt->AddPoint(*pointIt);
-                             (this->*fCalcSlope)(*lineIt);
-                             match = true;
-                             break;
-                         }
-                         // if this line consists of only one point so far, try again with different radius
-                         else if (lineIt->GetNPoints() == 1 and timeCondition and secondPointCondition)
-                         {
-                             KTDEBUG(stflog, "Trying initial-frequency-acceptance "<<fInitialFrequencyAcceptance);
-                             lineIt->AddPoint(*pointIt);
-                             (this->*fCalcSlope)(*lineIt);
-                             match = true;
-                             break;
-                         }
-                         // if not try next line
-                         else
-                         {
-                             ++lineIt;
-                         }
-                     }
-                 }//end of while loop. all existing lines were compared
-
-                 // if point was not picked up
-                 if (match == false)
-                 {
-                     //KTDEBUG(stflog, "Starting new line");
-
-                     KTSequentialLineData newLine;
-                     newLine.SetSlope( fInitialSlope );
-                     newLine.SetAcquisitionID( acqID );
-                     newLine.SetComponent( component );
-                     newLine.AddPoint(*pointIt);
-                     (this->*fCalcSlope)(newLine);
-                     fActiveLines.push_back(newLine);
-                     match = true;
-                 }
-             }
-         }
-         return true;
-     }
+        uint64_t acqID = kdTreeData.GetAcquisitionID();
+
+        // We need to be able to detect when we've moved from slice to slice
+        // So we define a threshold for delta-t, because time values might not be the exact same due to floating-point uncertainty.
+        // We'll use the bin width, which is tiny relative to the slice size and stride.
+        // 3-times the bin width should be large enough compared to uncertainty on the slice time and smaller than any reasonable stride.
```

Comment:
I thought all points from one slice had the same time in run. why is the time bin width different from the slice distance?

Reply:
In principle, yes, all points from one slice have the same time in run.  But there can be small differences between two different instances of what should be the same number stored because the actual floating-point number is stored as a binary representation that's not exact.  So if the time should be `1`, it might actually be represented as the equivalent of `0.9999999999...` in one place and `1.00...00001` in another place, and those would not test as equal.

The time bin width is 10 ns, whereas the slice length is 40960 ns.